src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1813,7 +1813,7 @@ trait Namers extends MethodSynthesis {
         }
 
         val newImport = treeCopy.Import(imp, expr1, selectors)
-        checkSelectors(newImport)
+        if (!context.unit.isJava) checkSelectors(newImport) // trust in javac
```

Comment:
Needs a test, because `checkSelectors` already has java checks. Isn't it helpful to know  if a name in Java means anything? I'd also object to the commented phrase. I would sit during that pledge, or probably even take a knee.

Maybe you and retronym are in the middle of a discussion and I'm lacking context.

Reply:
The context is that we're working on using `-Ypickle-java` to generate `.sig` files for jointly compiled `.java` sources which we can feed into a downstream `scalac` instance. This can shorten the critical path of a build graph (not because `javac` is slow, but because we can't start `javac` until `scalac` has delivered it `.class` files for upstream sub-projects.)

Anyway, under this mode we end up eagerly typechecking imports and method signatures of jointly compiled `.java` files, which has led us to fix up `Context#lookupSymbol` to respect Java name resolution rules around static classes.

We've also found cases where some check in `Typer` really isn't needed and we should just skip it.

Here, @hrhino discovered that changes in `lookupSymbol` unfortunately need to be duplicated here in `importSig` that there is some duplicated logic.

This check is a grey area between IMO. Skipping the check isn't ideal because:

  - If the imported name is actually referred to in the API of the class which we are pickling, we disagree with Javac. The failure mode could either a) by a "not found: type X" or b) could be that we resolve `X` to some other identifier contributed by another import.
  - As @som-snytt, it obscures the root cause of a build failure. In a mixed sources subproject built with `JavaThenScala`, `javac` is unlikely to run after a failure in `scalac` fails with case a) above.

Here's a test case where the import is material to the results of `-Ypickle-java`. (Where the reference to `I` in the body of `foo`, rather than its signature, this PR would have the desired effect.)

```java
package p1;

public class Test extends Base {}
class Base {
  static class I {}
}
```

```java
package p1;

import static p1.Test.I;

class Client {
  void foo(I i) {}
}
```

```
$ javac -d /tmp sandbox/Test.java sandbox/Client.java
$ scalac-ref 2.12.x -Ypickle-java sandbox/Test.java sandbox/Client.java -Ypickle-write /tmp/pickle
sandbox/Client.java:3: error: value I is not a member of object p1.Test
import static p1.Test.I;
       ^
sandbox/Client.java:6: error: not found: type I
  void foo(I i) {}
           ^
two errors found
$ scalac-pr 8735 -Ypickle-java sandbox/Test.java sandbox/Client.java -Ypickle-write /tmp/pickle


sandbox/Client.java:6: error: not found: type I
  void foo(I i) {}
           ^
one error found
```

Let me take a quick look at this area to see if there is an path to sharing the `Context#javaFindMember` logic with import checking/interpretation.

Reply:
> Needs a test, because checkSelectors already has java checks

To elaborate, if we were to just skip `checkSelector` for `.isJava`, we would also remove the `.isJava` exception from calling `checkNotRedundant`.

Reply:
Here's an alternative that uses `Context.javaFindMember` from `importedSymbol` and from `checkSelector`: https://github.com/scala/scala/pull/8755